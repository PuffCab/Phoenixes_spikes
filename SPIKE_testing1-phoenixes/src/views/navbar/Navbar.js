import React from "react";
//REVIEW[epic=snapshot, seq=1] 49- Create a component to test
function Navbar() {
  return (
    <nav>
      <li>Home</li>
      <li>About</li>
    </nav>
  );
}

export default Navbar;
