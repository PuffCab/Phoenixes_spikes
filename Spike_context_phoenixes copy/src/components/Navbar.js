import React, { useContext } from "react";
import { Button } from "react-bootstrap";
import { Link, NavLink, useNavigate } from "react-router-dom";
import { AuthContext } from "../context/AuthContext";

function Navbar() {
  const myStyle = {
    borderBottom: "solid 1px",
    marginBottom: "1rem",
  };
  const { user, setUser } = useContext(AuthContext);
  const goTo = useNavigate();

  const login = () => {
    setUser({
      name: "RaÃºl",
      email: "test@test.com",
    });
    goTo("/characters");
  };
  const logout = () => {
    setUser(null);
    goTo("/");
  };
  return (
    <>
      <nav style={myStyle}>
        <NavLink to="/">Home</NavLink> | {""}
        <Link to="/characters">Characters</Link> | {""}
        <NavLink
          style={({ isActive }) => {
            return isActive ? { color: "green" } : {};
          }}
          to="/about"
        >
          {({ isActive }) => {
            return isActive ? "About" : "not in About";
          }}
        </NavLink>{" "}
        | {""}
        <Link to="/seasons">Seasons</Link> | {""}
        {user ? (
          <Button variant="danger" onClick={logout}>
            Logout
          </Button>
        ) : (
          <Button variant="info" onClick={login}>
            Login
          </Button>
        )}
      </nav>
    </>
  );
}

export default Navbar;
