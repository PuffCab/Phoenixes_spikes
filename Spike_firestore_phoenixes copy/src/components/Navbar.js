import React, { useContext } from "react";
import { Button } from "react-bootstrap";
import { Link, NavLink, useNavigate } from "react-router-dom";
import { AuthContext } from "../context/AuthContext";

function Navbar() {
  const myStyle = {
    borderBottom: "solid 1px",
    marginBottom: "1rem",
  };
  const { user, setUser, logout } = useContext(AuthContext);
  const goTo = useNavigate();

  const login = () => {
    setUser({
      name: "RaÃºl",
      email: "test@test.com",
    });
    goTo("/characters");
  };
  const handleLogout = () => {
    logout();
    goTo("/");
  };
  return (
    <>
      <nav style={myStyle}>
        <NavLink to="/">Home</NavLink> | {""}
        <Link to="/characters">Characters</Link> | {""}
        <NavLink
          style={({ isActive }) => {
            return isActive ? { color: "green" } : {};
          }}
          to="/about"
        >
          {({ isActive }) => {
            return isActive ? "About" : "not in About";
          }}
        </NavLink>{" "}
        | {""}
        <Link to="/chat" className={!user ? "disabled" : ""}>
          Chat
        </Link>{" "}
        | {""}
        <Link to="/seasons">Seasons</Link> | {""}
        {user ? (
          <Link to="/login">Login</Link>
        ) : (
          <Link to="/register">Register</Link>
        )}
        | {""}
        {user ? (
          <Button variant="danger" onClick={handleLogout}>
            Logout
          </Button>
        ) : (
          <Button variant="info" onClick={login}>
            Login
          </Button>
        )}
      </nav>
    </>
  );
}

export default Navbar;
