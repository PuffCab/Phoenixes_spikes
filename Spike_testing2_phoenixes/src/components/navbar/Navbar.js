import React from "react";

function Navbar() {
  return (
    <nav>
      <ul>
        <li>one</li>
        <li>two</li>
        <li>three</li>
        <li>four</li>
      </ul>
    </nav>
  );
}

export default Navbar;
