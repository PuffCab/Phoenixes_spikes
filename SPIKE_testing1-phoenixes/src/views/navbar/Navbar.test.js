//REVIEW[epic=snapshot, seq=2] 50- We Install npm install -dev react-test-renderer from JEST https://jestjs.io/docs/tutorial-react
import renderer from "react-test-renderer";
import Navbar from "./Navbar";

//REVIEW[epic=snapshot, seq=3] 51- Create Test
test("Navbar component snapshot", () => {
  const component = renderer.create(<Navbar />);
  const domTree = component.toJSON();
  expect(domTree).toMatchSnapshot();
});

//REVIEW[epic=snapshot, seq=4] 52- IF we modify the component
// we UPDATE it with npm test -- -u
