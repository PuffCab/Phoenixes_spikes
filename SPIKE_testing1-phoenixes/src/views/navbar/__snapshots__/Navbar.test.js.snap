// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component snapshot 1`] = `
<nav>
  <li>
    Home
  </li>
  <li>
    About
  </li>
</nav>
`;
